src/infill/subDivCube.cpp

```diff
@@ -0,0 +1,279 @@
+            a.Y = center.y + relative_a.Y;
+            b.X = center.x + relative_b.X;
+            b.Y = center.y + relative_b.Y;
+            addLineAndCombine(*(directional_line_groups[idx]), a, b);
+            if (idx < 2)
+            {
+                rotatePoint120(relative_a);
+                rotatePoint120(relative_b);
+            }
+        }
+    }
+    for (int idx = 0; idx < 8; idx++) //!< draws the eight children
+    {
+        if (children[idx] != nullptr)
+        {
+            children[idx]->generateSubdivisionLines(z, result, directional_line_groups);
+        }
+    }
+    if (top_level) //!< copy directional groups into result, then free the directional groups
+    {
+        for (int temp = 0; temp < 3; temp++)
+        {
+            for (unsigned int idx = 0; idx < directional_line_groups[temp]->size(); idx++)
+            {
+                addLine((*directional_line_groups[temp])[idx][0], (*directional_line_groups[temp])[idx][1]);
+            }
+            delete directional_line_groups[temp];
+        }
+        free(directional_line_groups);
+    }
+}
+
+SubDivCube::SubDivCube(SliceMeshStorage& mesh, Point3& center, int depth)
+{
+    this->depth = depth;
+    this->center = center;
+
+    if (depth == 0) // lowest layer, no need for subdivision, exit.
+    {
+        return;
+    }
+    Point3 child_center;
+    coord_t radius = double(radius_multiplier * double(height[depth])) / 4.0 + radius_addition;
+
+    int child_nr = 0;
+    std::vector<Point3> rel_child_centers;
+    rel_child_centers.emplace_back(1, 1, 1); // top
+    rel_child_centers.emplace_back(-1, 1, 1); // top three
+    rel_child_centers.emplace_back(1, -1, 1);
+    rel_child_centers.emplace_back(1, 1, -1);
+    rel_child_centers.emplace_back(-1, -1, -1); // bottom
+    rel_child_centers.emplace_back(1, -1, -1); // bottom three
+    rel_child_centers.emplace_back(-1, 1, -1);
+    rel_child_centers.emplace_back(-1, -1, 1);
+    for (Point3 rel_child_center : rel_child_centers)
+    {
+        child_center = center + rotation_matrix.apply(rel_child_center * int32_t(side_length[depth] / 4));
+        if (isValidSubdivision(mesh, child_center, radius))
+        {
+            children[child_nr] = new SubDivCube(mesh, child_center, depth - 1);
+            child_nr++;
+        }
+    }
+}
+
+bool SubDivCube::isValidSubdivision(SliceMeshStorage& mesh, Point3& center, int64_t radius)
+{
+    int64_t distance2;
+    long int sphere_slice_radius2;//!< squared radius of bounding sphere slice on target layer
+    bool inside_somewhere = false;
+    bool outside_somewhere = false;
+    int inside;
+    double part_dist;//what percentage of the radius the target layer is away from the center along the z axis. 0 - 1
+    const long int layer_height = mesh.getSettingInMicrons("layer_height");
+    long int bottom_layer = (center.z - radius) / layer_height;
+    long int top_layer = (center.z + radius) / layer_height;
```

Comment:
I disagree. putting them inside the for loop makes it less readable and defining them before the for-loop helps the compiler in realizing that those values don't change over each iteration. This makes sure those values aren't recomputed each iteration, rather than leaving it up to the compiler.